modules/shapefile/src/lib/parse-dbf.js

```diff
@@ -0,0 +1,131 @@
+/* global TextDecoder */
+
+const LITTLE_ENDIAN = true;
+
+export default function parseDbf(arrayBuffer) {
+  // Global header
+  const globalHeaderView = new DataView(arrayBuffer, 0, 32);
+  const header = parseHeader(globalHeaderView);
+  const {headerLength, recordLength, nRecords} = header;
+
+  // Row headers
+  // TODO: use code page (.cpg) for encoding. Default to ASCII or UTF-8?
+  const encoding = 'utf-8';
+  // Requires Node 11+
+  const textDecoder = new TextDecoder(encoding);
+  const colHeaderView = new DataView(arrayBuffer, 32, headerLength - 32);
+  const colHeaders = parseColumnHeaders(colHeaderView, textDecoder);
+
+  // Not exactly sure why start offset needs to be headerLength + 1?
+  // parsedbf uses ((colHeaders.length + 1) << 5) + 2;
+  const recordsView = new DataView(arrayBuffer, headerLength + 1);
+  return parseRows(recordsView, colHeaders, nRecords, recordLength, textDecoder);
+}
+
+function parseHeader(headerView) {
+  // Last updated date
+  const year = headerView.getUint8(1) + 1900;
+  const month = headerView.getUint8(2);
+  const day = headerView.getUint8(3);
+
+  // Number of records in data file
+  const nRecords = headerView.getUint32(4, LITTLE_ENDIAN);
+  // Length of header in bytes
+  const headerLength = headerView.getUint16(8, LITTLE_ENDIAN);
+  // Length of each record
+  const recordLength = headerView.getUint16(10, LITTLE_ENDIAN);
+  // Not sure if this is usually set
+  const languageDriver = headerView.getUint8(29);
+
+  return {
+    year,
+    month,
+    day,
+    nRecords,
+    headerLength,
+    recordLength,
+    languageDriver
+  };
+}
+
+function parseColumnHeaders(view, textDecoder) {
+  // NOTE: this might overestimate the number of fields if the "Database
+  // Container" container exists and is included in the headerLength
+  const nFields = (view.byteLength - 1) / 32;
+  const fields = [];
+  let offset = 0;
+  for (let i = 0; i < nFields; i++) {
+    // Better way to read text?
+    const name = textDecoder
+      .decode(view.buffer.slice(view.byteOffset + offset, view.byteOffset + offset + 11))
```

Comment:
For performance i suspect it is faster to just create a Uint8Array view of those bytes rather than create an ArrayBuffer copy?